modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3447,13 +3501,12 @@ else if (cntr == maxCntr.cnt)
                 Long localHistCntr = localReserved.get(p);
 
                 if (localHistCntr != null && maxCntrObj.nodes.contains(cctx.localNodeId())) {
-                    if (localHistCntr <= minCntr) {
+                    Long ceilingMinReserved = minCntrs.ceiling(localHistCntr);
 
-                        partHistSuppliers.put(cctx.localNodeId(), top.groupId(), p, localHistCntr);
+                    if (ceilingMinReserved != null) {
+                        partHistSuppliers.put(cctx.localNodeId(), top.groupId(), p, ceilingMinReserved);
```

Comment:
Why not partHistSuppliers.put(cctx.localNodeId(), top.groupId(), p, localHistCntr); ?

Reply:
We decided stay it as is, because this change can lead to reserve mode history than now.